astropy/coordinates/angle_utilities.py

```diff
@@ -184,33 +202,62 @@ def p_colon(p):
             '''
             colon : sign UINT COLON ufloat
                   | sign UINT COLON UINT COLON ufloat
+                  | sign UINT COLON ufloat dir
```

Comment:
I think the way to prohibit having both sign and direction is to simply remove `sign` in this line and the next. Then, you do not have to do the `if` statement just below. Note that this assumes we also want to forbid `+10d20mW`, which I think is good.

Reply:
No, I can't do that because no matter what direction will come into the picture because in grammar it is defined if no sign is present assume it to be 1.0
P.S. according to grammar defined +10d20mW is same as 10d20mW.

Reply:
I think we can avoid that if we put the ones with the direction first, i.e., the first entry would be
```
UINT COLON ufloat dir
```